### DISCUSSION TITLE
Model viewer now working in Server Side Rendering

### DISCUSSION THREAD

**Discussion by RRonier** [google/model-viewer#2877] [Q&A]
Hello community,
I'm trying to use model viewer in a NextJs project, but when I try to run my webapp it throws me an error:
ReferenceError: window is not defined
at ....\node_modules\@google\model-viewer\dist\model-viewer-umd.js
How can I fix this? Is an error I can fix or is a bug?
Thank you very much

**Comment by RRonier** [comment]
@timmmeh @elalish

**Reply by elalish** [reply]
model-viewer is a web component, and as such is very dependent on the DOM, which is not present in  Node.js. You'll need to use Puppeteer or something similar to get headless Chrome running where model-viewer can operate. We automate our fidelity tests this way; you might want to look at our `render-fidelity-tools` package.

**Comment by Yuripetusko** [comment]
Simply import a component with model viewer using nextjs dynamic import with ssr: false option

**Reply by Poem7200** [reply]
how to add ssr:false option, could you give an example? I also come up with the same problem, but the demo below can not solve it…
https://codesandbox.io/s/github/ajmhyd/google-model-viewer-nextjs?file=/components/Model.js:140-238

**Reply by Yuripetusko** [reply]
We use nextjs

```

const ComponentWithModelViewer = dynamic(
  () => import('components/componentWithModelViewer'),
  {
    ssr: false,
  },
);
```

**Reply by Poem7200** [reply]
Thanks, finally I use another way to import this dependency, just add useEffect hook to import @google/model-viewer after mount

`useEffect(() => {
    import('@google/model-viewer').catch(console.error);
  }, []);`

**Reply by tomcatmew** [reply]
Can confirm @Poem7200 's solution works.
Life savier ! Thanks !

**Reply by guvargas** [reply]
This also work for older versions of next, with the error being something like 
```
⨯ node_modules\@google\model-viewer\lib\constants.js (18:36) @ navigator
⨯ ReferenceError: navigator is not defined
```
